tools/odbc/test/tests/col_attribute/col_attribute.cpp

```diff
@@ -0,0 +1,131 @@
+	REQUIRE(num_cols == 6);
+
+	// Loop through the columns
+	for (int i = 1; i <= num_cols; i++) {
+		char buffer[64];
+
+		// Get the column label
+		EXECUTE_AND_CHECK("SQLColAttribute", SQLColAttribute, hstmt, i, SQL_DESC_LABEL, buffer, sizeof(buffer), nullptr,
+		                  nullptr);
+
+		// Get the column name and base column name
+		char col_name[64];
+		char base_col_name[64];
+		EXECUTE_AND_CHECK("SQLColAttribute", SQLColAttribute, hstmt, i, SQL_DESC_NAME, col_name, sizeof(col_name),
+		                  nullptr, nullptr);
+		EXECUTE_AND_CHECK("SQLColAttribute", SQLColAttribute, hstmt, i, SQL_DESC_BASE_COLUMN_NAME, base_col_name,
+		                  sizeof(base_col_name), nullptr, nullptr);
+		REQUIRE(STR_EQUAL(col_name, base_col_name));
+		switch (i) {
+		case 1:
+			REQUIRE(STR_EQUAL(buffer, "intcol"));
+			REQUIRE(STR_EQUAL(col_name, "intcol"));
+			REQUIRE(STR_EQUAL(base_col_name, "intcol"));
+			break;
+		case 2:
+			REQUIRE(STR_EQUAL(buffer, "textcol"));
+			REQUIRE(STR_EQUAL(col_name, "textcol"));
+			REQUIRE(STR_EQUAL(base_col_name, "textcol"));
+			break;
+		case 3:
+			REQUIRE(STR_EQUAL(buffer, "varcharcol"));
+			REQUIRE(STR_EQUAL(col_name, "varcharcol"));
+			REQUIRE(STR_EQUAL(base_col_name, "varcharcol"));
+			break;
+		case 4:
+			REQUIRE(STR_EQUAL(buffer, "empty_varchar_col"));
+			REQUIRE(STR_EQUAL(col_name, "empty_varchar_col"));
+			REQUIRE(STR_EQUAL(base_col_name, "empty_varchar_col"));
+			break;
+		case 5:
+			REQUIRE(STR_EQUAL(buffer, "varchar5col"));
+			REQUIRE(STR_EQUAL(col_name, "varchar5col"));
+			REQUIRE(STR_EQUAL(base_col_name, "varchar5col"));
+			break;
+		case 6:
+			REQUIRE(STR_EQUAL(buffer, "CAST('5 days' AS INTERVAL)"));
+			REQUIRE(STR_EQUAL(col_name, "CAST('5 days' AS INTERVAL)"));
+			REQUIRE(STR_EQUAL(base_col_name, "CAST('5 days' AS INTERVAL)"));
+			break;
+		}
+
+		// Get the column octet length
+		CheckInteger(hstmt, 0, SQL_DESC_OCTET_LENGTH);
+
+		// Get the column type name
+		EXECUTE_AND_CHECK("SQLColAttribute", SQLColAttribute, hstmt, i, SQL_DESC_TYPE_NAME, buffer, sizeof(buffer),
+		                  nullptr, nullptr);
+		switch (i) {
+		case 1:
+			REQUIRE(STR_EQUAL(buffer, "INT32"));
+			break;
+		case 2:
+		case 3:
+		case 4:
+		case 5:
+			REQUIRE(STR_EQUAL(buffer, "VARCHAR"));
+			break;
+		case 6:
+			REQUIRE(STR_EQUAL(buffer, "INTERVAL"));
+			break;
+		}
+	}
+
+	// SQLColAttribute should fail if the column number is out of bounds
+	ret = SQLColAttribute(hstmt, 7, SQL_DESC_TYPE_NAME, nullptr, 0, nullptr, nullptr);
+	REQUIRE(ret == SQL_ERROR);
+
+	// Create table and retrieve base table name using SQLColAttribute, should fail because the statement is not a
+	// SELECT
+	EXECUTE_AND_CHECK("SQLExecDirect", SQLExecDirect, hstmt,
+	                  ConvertToSQLCHAR("CREATE TABLE test (a INTEGER, b INTEGER)"), SQL_NTS);
+	ExpectError(hstmt, SQL_DESC_BASE_TABLE_NAME);
+
+	// Prepare a statement and call SQLColAttribute, succeeds but is undefined
```

Comment:
`succeeds but is undefined`: what does this mean?

Reply:
It succeeds, but because the statement is prepared but not yet executed it can create undefined behavior. 
Related to PR #9316